verchk.go

```diff
@@ -73,31 +70,8 @@ func (v *versionChecker) generateClientUUID() (string, error) {
 }
 
 func (v *versionChecker) writeClientUUID() (string, error) {
-	filename := filepath.Join(v.path, versionCheckerFilename)
-	if _, err := os.Stat(filename); err != nil {
-		if os.IsNotExist(err) {
-			fh, err := os.Create(filename)
-			if err != nil {
-				return "", err
-			}
-			defer fh.Close()
-
-			id, err := v.generateClientUUID()
-			if err != nil {
-				return "", err
-			}
-
-			if _, err := fh.WriteString(id); err != nil {
-				return "", err
-			}
-
-			return id, nil
-		} else {
-			return "", err
-		}
-	}
-
-	fh, err := os.Open(filename)
+	filename := filepath.Join(v.path, v.idfile)
```

Comment:
Hmm. Okay, so, imagine that I, as a user, write `-uuid-file=/tmp/uuid`. We're going to then try to use the file name `/path/to/data//tmp/uuid`.

I think we should somehow handle the case where the path name is absolute. Note that it's not strictly portable to check the first character for being `/`, but I don't remember how to do it correctly in Go.

Reply:
`filepath.Join("/path/to/data/", "/tmp/uuid")` should give you `"/path/to/data/tmp/uuid"` (I might be missing something here)